src/index.ts

```diff
@@ -17,4 +17,14 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-export { scan } from './scan';
+import { scan } from './scan';
+import { ScanOptions } from './types';
+
+export { scan };
+
+export function customScanner(scanOptions: ScanOptions) {
+  return scan({
+    ...scanOptions,
+    localScannerCli: true,
+  });
+}
```

Comment:
I think I am missing something here, why do we need a new export function to handle the `localScannerCli`?  

It seems to align closer to the original implementation here right? 

https://github.com/SonarSource/sonar-scanner-npm/blob/master/src/sonar-scanner-executable.js#L42C7-L42C19

looking closer, they also support passing an explicit callback too
https://github.com/SonarSource/sonar-scanner-npm/blob/master/src/index.js#L44C1-L51C2

we could also migrate some of the unit tests at this point to make sure they would pass

Reply:
> It seems to align closer to the original implementation here right?

Yep it's to match the previously exported scan function documented in the README

The others weren't documented so IMO we can drop them(?) @vdiez wdyt?

Reply:
we will address this point in 
https://sonarsource.atlassian.net/browse/SCANNPM-8

Reply:
I agree with cleaning the exports, but we should document what is being dropped. Also, currently the default export accepts a callback, so I would suggest to keep it, and create an async version aside of that one (currently named `scan`). `customScanner` uses currently callback, with this change becomes async.